drivers/gles2/rasterizer_canvas_gles2.h

```diff
@@ -31,118 +31,254 @@
+private:
+	// legacy codepath .. to remove after testing
+	void _canvas_render_item(Item *p_ci, RenderItemState &r_ris);
+	_FORCE_INLINE_ void _canvas_item_render_commands(Item *p_item, Item *p_current_clip, bool &r_reclip, RasterizerStorageGLES2::Material *p_material);
+
+	// high level batch funcs
+	void canvas_render_items_implementation(Item *p_item_list, int p_z, const Color &p_modulate, Light *p_light, const Transform2D &p_base_transform);
+	void render_joined_item(const BItemJoined &p_bij, RenderItemState &r_ris);
+	void join_items(Item *p_item_list, int p_z, const Color &p_modulate, Light *p_light, const Transform2D &p_base_transform);
+	bool try_join_item(Item *p_ci, RenderItemState &r_ris, bool &r_batch_break);
+	void render_joined_item_commands(const BItemJoined &p_bij, Item *p_current_clip, bool &r_reclip, RasterizerStorageGLES2::Material *p_material);
+	void render_batches(Item::Command *const *p_commands, int p_first_item_ref_id, Item *p_current_clip, bool &r_reclip, RasterizerStorageGLES2::Material *p_material);
+	bool prefill_joined_item(FillState &r_fill_state, int &r_command_start, Item *p_item, Item *p_current_clip, bool &r_reclip, RasterizerStorageGLES2::Material *p_material);
+	void flush_render_batches(Item *p_item, Item *p_current_clip, bool &r_reclip, RasterizerStorageGLES2::Material *p_material);
+
+	// low level batch funcs
+	void _batch_translate_to_colored();
+	_FORCE_INLINE_ int _batch_find_or_create_tex(const RID &p_texture, const RID &p_normal, bool p_tile, int p_previous_match);
+	RasterizerStorageGLES2::Texture *_get_canvas_texture(const RID &p_texture) const;
+	void _batch_upload_buffers();
+	void _batch_render_rects(const Batch &p_batch, RasterizerStorageGLES2::Material *p_material);
+	BatchVertex *_batch_vertex_request_new() { return bdata.vertices.request(); }
+	Batch *_batch_request_new(bool p_blank = true);
+
+	bool _detect_batch_break(Item *p_ci);
+	void _software_transform_vertex(BatchVector2 &r_v, const Transform2D &p_tr) const;
+	void _software_transform_vertex(Vector2 &r_v, const Transform2D &p_tr) const;
+	TransformMode _find_transform_mode(bool p_use_hardware_transform, const Transform2D &p_tr, Transform2D &r_tr) const;
+
+	// light scissoring
+	bool _light_find_intersection(const Rect2 &p_item_rect, const Transform2D &p_light_xform, const Rect2 &p_light_rect, Rect2 &r_cliprect) const;
+	bool _light_scissor_begin(const Rect2 &p_item_rect, const Transform2D &p_light_xform, const Rect2 &p_light_rect) const;
+	void _calculate_scissor_threshold_area();
 
+public:
 	void initialize();
-	void finalize();
-
-	virtual void draw_window_margins(int *black_margin, RID *black_image);
-
 	RasterizerCanvasGLES2();
 };
 
+//////////////////////////////////////////////////////////////
+
+inline void RasterizerCanvasGLES2::_software_transform_vertex(BatchVector2 &r_v, const Transform2D &p_tr) const {
```

Comment:
use _FORCE_INLINE_ macro instead of inline, to ensure the compiler will always inline it

Reply:
Done.